### PR TITLE
Adding upstream chart

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/thg/issues/114

### Testing

Description on how {APP-NAME} can be tested.

- [ ] fresh install works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM
- [ ] upgrade from previous version works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM

#### Other testing

Description of features to additionally test for {APP-NAME} installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Comment by ubergesundheit** [comment]
Multiple things:
- Looks like you executed the subtree against some `edge-xxxx` release or some commit in the upstream main branch. I strongly suggest to use the latest `stable-2.11.1` tag from upstream to have all apps (cni, main app, target and this one) aligned on the same chart version
- Looks like the partials subchart is also required.

**Comment by pipo02mix** [comment]
@ubergesundheit anything else?

File: .abs/main.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 replace-chart-version-with-git: true
 generate-metadata: true
-chart-dir: ./helm/{APP-NAME}
+chart-dir: ./helm/linkerd2-multicluster-source-app
 destination: ./build
```
Comment (ubergesundheit) [inline .abs/main.yaml:4]:
```suggestion
destination: ./build
# until we support custom template helper filenames
disable-giantswarm-helm-validator: true
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -8,10 +8,10 @@ workflows:
       - architect/push-to-app-catalog:
           context: "architect"
           # executor: "app-build-suite" # uncomment this if you want automatic metadata generation and helm chart linting
```
Comment (ubergesundheit) [inline .circleci/config.yml]:
```suggestion
          executor: "app-build-suite"
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -8,10 +8,10 @@ workflows:
...
-          name: "package and push {APP-NAME} chart"
+          name: "package and push linkerd2-multicluster-source-app chart"
           app_catalog: "giantswarm-playground-catalog"
           app_catalog_test: "giantswarm-playground-test-catalog"
```
Comment (ubergesundheit) [inline .circleci/config.yml]:
```suggestion
          app_catalog: "giantswarm-catalog"
          app_catalog_test: "giantswarm-test-catalog"
```

File: helm/linkerd2-multicluster-source-app/values.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# -- Docker image for the Service mirror component (uses the Linkerd controller
+# image)
+controllerImage: cr.l5d.io/linkerd/controller
```
Comment (ubergesundheit) [inline helm/linkerd2-multicluster-source-app/values.yaml]: we should use the retagged giantswarm container image

File: helm/linkerd2-multicluster-source-app/values.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+# -- Tag for the Service Mirror container Docker image
+controllerImageVersion: linkerdVersionValue
```
Comment (ubergesundheit) [inline helm/linkerd2-multicluster-source-app/values.yaml]: I think this needs to be something valid like `stable-2.11.2` or `2.11.2`

Unsure about the exact value

File: helm/linkerd2-multicluster-source-app/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+  Warning: The purpose of this chart is just to support the `linkerd
+  multicluster link` CLI command, which also produces the
+  `cluster-credentials` secret and the Link CR, which are not found in this
+  chart. Therefore this chart is not a replacement for that command, and
+  shouldn't be used as-is unless you really know what you're doing ;-)
+kubeVersion: ">=1.17.0-0"
+icon: https://linkerd.io/images/logo-only-200h.png
+name: "linkerd-multicluster-link"
+version: 0.1.0
```
Comment (ubergesundheit) [inline helm/linkerd2-multicluster-source-app/Chart.yaml:15]:
```suggestion
version: 0.1.0
restrictions:
  clusterSingleton: true
  fixedNamespace: linkerd-multicluster
annotations:
  application.giantswarm.io/team: team-cabbage
```

**Comment by ubergesundheit** [comment]
I've done the circle ci setup, please push a commit to trigger CI :)

File: helm/linkerd2-multicluster-source-app/values.yaml (hunk 4 above)
Comment (pipo02mix) [inline helm/linkerd2-multicluster-source-app/values.yaml]:
```suggestion
controllerImage: quay.io/giantswarm/linkerd2-controller
```

File: helm/linkerd2-multicluster-source-app/values.yaml (hunk 5 above)
Comment (pipo02mix) [inline helm/linkerd2-multicluster-source-app/values.yaml]:
```suggestion
controllerImageVersion: stable-2.11.2
```

**Comment by pipo02mix** [comment]
@ubergesundheit I added some exceptions for kube linter but it is not taken into account
helm/linkerd2-multicluster-source-app/kube-linter.yaml

File: helm/linkerd2-multicluster-source-app/kube-linter.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+checks:
```
Comment (ubergesundheit) [inline helm/linkerd2-multicluster-source-app/kube-linter.yaml]: I think the file needs a leading dot in its name (`.kube-linter.yaml`)

**Comment by ubergesundheit** [comment]
I would also suggest du update the `architect-orb` version in the circleCI config file to the latest version (4.23.0)

**Comment by pipo02mix** [comment]
CI green!

**Review by ubergesundheit — APPROVED** [review]
LGTM, remember to not Squash merge this PR using GitHub. Use "Create merge commit", otherwise the subtree information will be destroyed

**Pull request merged**